### ISSUE TITLE
Typedef of template instance fails

### ISSUE THREAD

**Issue by senier** [Componolit/Cappulada#64]
When typedef'ing a template instance to a new name

```
typedef Container<int> Int_Container;
```

the following error occurs:

```
Traceback (most recent call last):
  File "./tests/testconvert.py", line 367, in test_template_typedef
    CXX("tests/data/test_template_typedef.h").ToIR(project="Capdpa")
  File "/home/alex/Devel/src/capdpa/capdpa/cxx.py", line 391, in ToIR
    children = self.__convert_children(self.translation_unit.cursor.get_children()),
  File "/home/alex/Devel/src/capdpa/capdpa/cxx.py", line 342, in __convert_children
    children.append(self.__convert_typedef(cursor))
  File "/home/alex/Devel/src/capdpa/capdpa/cxx.py", line 291, in __convert_typedef
    resolved = self.__convert_type([], children[0].type)
  File "/home/alex/Devel/src/capdpa/capdpa/cxx.py", line 230, in __convert_type
    raise NotImplementedError("Unsupported type: {} (from {})".format(type_cursor.kind, type_cursor.spelling))
NotImplementedError: Unsupported type: TypeKind.INVALID (from )
```

Test case is in `tests/testconvert.py` (`EXCLUDE_test_template_typedef`).

**Comment by jklmnn** [comment]
I think already fixed this in https://github.com/Componolit/Cappulada/commit/a927a9071268d170de2c3bebdf58350fd8e418eb. Can you check if this commit is available on your current branch? GitHub tells me it's on issue_32 and issue_35.

**Comment by senier** [comment]
I rebased https://github.com/Componolit/Cappulada/commit/a927a9071268d170de2c3bebdf58350fd8e418eb, but the error persists:

```
ERROR: test_template_typedef (__main__.Parser)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./tests/testconvert.py", line 377, in test_template_typedef
    self.check(CXX("tests/data/test_template_typedef.h").ToIR(project="Capdpa"), expected)
  File "/home/alex/Devel/src/capdpa/capdpa/cxx.py", line 391, in ToIR
    children = self.__convert_children(self.translation_unit.cursor.get_children()),
  File "/home/alex/Devel/src/capdpa/capdpa/cxx.py", line 342, in __convert_children
    children.append(self.__convert_typedef(cursor))
  File "/home/alex/Devel/src/capdpa/capdpa/cxx.py", line 291, in __convert_typedef
    resolved = self.__convert_type([], children[0].type)
  File "/home/alex/Devel/src/capdpa/capdpa/cxx.py", line 230, in __convert_type
    raise NotImplementedError("Unsupported type: {} (from {})".format(type_cursor.kind, type_cursor.spelling))
NotImplementedError: Unsupported type: TypeKind.INVALID (from )
```

Am I missing any commit?

**Comment by jklmnn** [comment]
Hmm, afaik this was the commit to fix this, but I did multiple commits relating to different `INVALID` messages so it could be that some other, seemingly unrelated commit is required. If you run the tests on branch issue_32 this error should not appear. Yet I have to take a deeper look into that.

**Comment by jklmnn** [comment]
Fixed by bfba07bef4452c2f39cb9337a35a5ad4f9d3e3db.
